### ISSUE TITLE
[Remote-SSH Bug]: "remote.SSH.connectTimeout" overrides `ConnectTimeout` from config file

### ISSUE THREAD

**Issue by jeanp413** [microsoft/vscode-remote-release#8519]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### -> Description

The "remote.SSH.connectTimeout" setting value overrides `ConnectTimeout` from config file

### Expected Behavior

I expect `ConnectTimeout` from config file to have precedence over "remote.SSH.connectTimeout" as I have some host that require more time but in general I'm fine with the default value in vscode when no `ConnectTimeout` value is specified in config file

**Comment by akosyakov** [comment]
To add: it should also respect when ConnectTimeout is configured as 0, meaning to disable it.

**Comment by akosyakov** [comment]
If it would be helpful one can test using the following shell script
```bash
#!/bin/bash

# Variables for the host and port from SSH configuration
HOST="$1"   # SSH will pass the host as the first argument
PORT="$2"   # SSH will pass the port as the second argument

# Sleep for 1 minute while printing a line every second
for i in {1..60}; do
    echo "Sleeping... $i seconds elapsed"
    sleep 1
done

# Now use nc to connect to the given host and port
echo "Attempting to connect to $HOST on port $PORT"
nc $HOST $PORT
```

And the configure the host in ssh config like:
```
Host my-host
  ProxyCommand /Users/my-user/.ssh/pause.sh %h %p
  ConnectTimeout 0
```

Given default Remote SSH settings, it fails right now like:
<img width="1451" alt="Screenshot 2024-09-11 at 11 02 07" src="https://github.com/user-attachments/assets/c586267c-d738-45af-bbd1-3bff0d1456e4">

Compare without Remote SSH:
<img width="902" alt="Screenshot 2024-09-11 at 11 12 10" src="https://github.com/user-attachments/assets/41fa4e2f-c800-43dc-af27-845b16516fd6">

**Comment by joshspicer** [comment]
This is shipped in the latest pre-release build (`v0.115.2024092017 (pre-release)`).  I've tested with your repro steps and it is working well for me!  Please give it a go and let me know if you have any feedback.

**Comment by akosyakov** [comment]
@joshspicer thank you! we verified on our side and it seems to work well
